src/urllib3/connectionpool.py

```diff
@@ -258,7 +258,16 @@ def _get_conn(self, timeout: Optional[float] = None) -> HTTPConnection:
             raise ClosedPoolError(self, "Pool is closed.")
 
         try:
-            conn = self.pool.get(block=self.block, timeout=timeout)
+            if self.block and timeout is None:
+                # Don't wait and hang forever. Give the developer a hit if they forget to release the connection.
+                while True:
+                    try:
+                        conn = self.pool.get(block=self.block, timeout=60)
+                        break
+                    except queue.Empty:
+                        log.debug('Still waiting for available connections. Did you forget to release them?')
```

Comment:
So in production, if the pool (or pools) is just particularly active, we could spam users' logs? I get that this is debug, but some people do enable urllib3 debug logs purposefully. What happens if there are multiple pools having similar issues? There's no indication as to which pool is having trouble and there's no indication as to how to fix this if you're using a library built atop urllib3 because it's not actionable.